src/main/java/echoserver/EchoServer.java

```diff
@@ -1,18 +1,33 @@
 package echoserver;
 
+import java.io.IOException;
+import java.util.concurrent.Executor;
+
 public class EchoServer {
     private final Listenable listener;
     private final Echoable echoer;
+    private final Executor executor;
+    private final Loggable logger;
 
-    public EchoServer(Listenable listener, Echoable echoer) {
+    public EchoServer(Listenable listener, Executor executor, Echoable echoer, Loggable logger) {
         this.listener = listener;
         this.echoer = echoer;
+        this.executor = executor;
+        this.logger = logger;
     }
 
     public void start() throws Exception {
         Connection socketConnection;
         while((socketConnection = this.listener.listen()) != null) {
-            this.echoer.echo(socketConnection);
+            Connection finalSocketConnection = socketConnection;
+            Runnable runnable = () -> {
+                try {
+                    this.echoer.echo(finalSocketConnection);
+                } catch (IOException e) {
+                    this.logger.log("Failed to echo.");
```

Comment:
Same question here regarding catching the exception.

Reply:
I'm not sure what to do with this exception - I can't really close anything.  Besides logging the error, is there anything else to do?

Reply:
I could create a custom exception as I described above.  What do you suggest?